src/components/Composer/index.js

```diff
@@ -106,363 +104,352 @@ const defaultProps = {
-        if (this.props.forwardedRef && _.isFunction(this.props.forwardedRef)) {
-            this.props.forwardedRef(this.textInput);
-        }
-
-        // There is no onPaste or onDrag for TextInput in react-native so we will add event
-        // listeners here and unbind when the component unmounts
-        if (this.textInput) {
-            this.textInput.addEventListener('wheel', this.handleWheel);
-
-            // we need to handle listeners on navigation focus/blur as Composer is not unmounting
-            // when navigating away to different report
-            this.unsubscribeFocus = this.props.navigation.addListener('focus', () => document.addEventListener('paste', this.handlePaste));
-            this.unsubscribeBlur = this.props.navigation.addListener('blur', () => document.removeEventListener('paste', this.handlePaste));
-
-            // We need to add paste listener manually as well as navigation focus event is not triggered on component mount
-            document.addEventListener('paste', this.handlePaste);
-        }
-    }
-
-    componentDidUpdate(prevProps) {
-        if (!prevProps.shouldClear && this.props.shouldClear) {
-            this.textInput.clear();
-            // eslint-disable-next-line react/no-did-update-set-state
-            this.setState({numberOfLines: 1});
-            this.props.onClear();
-        }
-
-        if (
-            prevProps.value !== this.props.value ||
-            prevProps.defaultValue !== this.props.defaultValue ||
-            prevProps.isComposerFullSize !== this.props.isComposerFullSize ||
-            prevProps.windowWidth !== this.props.windowWidth ||
-            prevProps.numberOfLines !== this.props.numberOfLines
-        ) {
-            this.updateNumberOfLines();
-        }
-
-        if (prevProps.selection !== this.props.selection) {
-            // eslint-disable-next-line react/no-did-update-set-state
-            this.setState({selection: this.props.selection});
-        }
-    }
+    // If there is a space or new line, return the substring up to the space or new line
+    return substr.substring(0, spaceIndex);
+};
 
-    componentWillUnmount() {
-        if (!this.textInput) {
+// Enable Markdown parsing.
+// On web we like to have the Text Input field always focused so the user can easily type a new chat
+function Composer({
+    value = undefined,
+    defaultValue = undefined,
+    maxLines = -1,
+    onKeyPress,
+    style = null,
+    shouldClear = false,
+    autoFocus = false,
+    translate,
+    isFullComposerAvailable = false,
+    shouldCalculateCaretPosition = false,
+    numberOfLines: numberOfLinesProp = undefined,
+    isDisabled = false,
+    forwardedRef = null,
+    navigation,
+    onClear = () => {},
+    onPasteFile = () => {},
+    onSelectionChange = () => {},
+    onNumberOfLinesChange = () => {},
+    setIsFullComposerAvailable = () => {},
+    checkComposerVisibility = () => false,
+    selection: selectionProp = {start: 0, end: 0},
```

Comment:
Remove default values as they are defined in default props, keep only destructed props without values

Reply:
👍🏻